htdocs/compta/bank/releve.php

```diff
@@ -185,163 +185,120 @@
-                        $payment->fetch($val['url_id']);
+                        $payment->fetch('','',$objd->rowid);
                         $arraybill = $payment->getBillsArray();
+
                         if (is_array($arraybill) && count($arraybill) > 0)
                         {
+                            $facturestatic=new Facture($db);
                             foreach ($arraybill as $billid)
-                            {
+                            {                             
                                 $facturestatic->fetch($billid);
-                                $subdir = get_exdir($facturestatic->id, 2, 0, 0, $facturestatic, 'invoice');
-
+                                $subdir = $facturestatic->getFilesDir();
                                 $arrayofinclusion=array();              // TODO Find a way to get doc ODT or other
-                                // TODO Use get_exdir
-                                $arrayofinclusion[]=preg_quote($facturestatic->ref.'.pdf','/');
-                                $listoffiles = dol_dir_list($conf->facture->dir_output.$subdir,'all',1,implode('|',$arrayofinclusion),'\.meta$|\.png','date',SORT_DESC,0,true);
-                                // build list of files with full path
-                                $files = array();
-                                foreach($listoffiles as $filefound)
+                                $arrayofinclusion[]=".*\.odt";
+                                $arrayofinclusion[]=".*\.pdf";
+                                $listoffiles = dol_dir_list($subdir,'all',1,implode('|',$arrayofinclusion),'\.meta$|\.png','date',SORT_DESC,0,true);
+                                foreach($listoffiles as $srcfileobj)
                                 {
-                                    if (strstr($filefound["name"],$facturestatic->ref))
-                                    {
-                                        $files[] = $uploaddir.'/'.$facturestatic->ref.'/'.$filefound["name"];
-                                        break;
-                                    }
-                                }
-                                /*var_dump($files);*/
-                                //var_dump($listoffiles);
-                                foreach($listoffiles as $key => $srcfileobj)
-                                {
-                                    $srcfile = $srcfileobj['fullname'];
-                                    $destfile = $outdirinvoices.'/'.$srcfileobj['name'];
-                                    //var_dump($srcfile.' - '.$destfile);
-                                    dol_copy($srcfile, $destfile);
+                                     $upload_files[]=$srcfileobj;
                                 }
                             }
                         }
                         break;
-                    case "payment_supplier":
+                    case "(SupplierInvoicePayment)":
```

Comment:
rely on string "(SupplierInvoicePayment)" is not possible because the string may have been replaced with any other text by the user.
Currently type of line can be defined from content of    $links = $object->get_url($objd->rowid);